### PR TITLE
Fix qt kits

### PR DESCRIPTION
**Pull request by rhubert**

### CODE REVIEW

File: pym/bob/generators/QtCreatorGenerator.py [hunk 1]
```diff
@@ -169,28 +171,36 @@ def generateQtProject(package, destination, updateOnly, projectName, includeDirs
...
-        if os.path.exists(i):
-            for root, directories, filenames in os.walk(i):
-                hList.append(os.path.join(i,root))
+                if not hasInclude and include.match(filename):
+                    hasInclude = True
+            if hasInclude:
+                oldPath = ""
+                # need to recursively add all directories from the include up to cwd to get includes like
+                # <a/b/c.h> resolved even if there is no include in 'a'
+                root = root[len(os.getcwd()):]
```
Comment (jkloetzke) [inline pym/bob/generators/QtCreatorGenerator.py]: Isn't `os.getcwd()` wrong here? The base directory is `newPath` which could be a totally different directory if the user specified "--destination"...

File: pym/bob/generators/QtCreatorGenerator.py [hunk 2]
```diff
@@ -169,28 +171,36 @@ def generateQtProject(package, destination, updateOnly, projectName, includeDirs
...
+                    if not newPath in hList:
+                        hList.append(newPath)
+                    oldPath = os.path.join(oldPath, path)
 
     # use default kit "Desktop" if no kit is given
     if kit is None:
-        kit = "Desktop"
+        _kit = re.compile(r".*Desktop.*")
+    else:
+        _kit = re.compile(r""+kit)
```
Comment (jkloetzke) [inline pym/bob/generators/QtCreatorGenerator.py]: Might not be an error. Is it intended that the user gives a regex in "--kit"?

Comment (rhubert) [inline pym/bob/generators/QtCreatorGenerator.py]: yes it was my intend to allow a regex for --kit. Even if this is probably never used...

File: pym/bob/generators/QtCreatorGenerator.py (hunk 1 above)
Comment (rhubert) [inline pym/bob/generators/QtCreatorGenerator.py]: :+1: you're right...

**Comment by rhubert** [comment]
Please do not merge. I'd like to do some more testing tomorrow...

File: pym/bob/generators/QtCreatorGenerator.py (hunk 2 above)
Comment (jkloetzke) [inline pym/bob/generators/QtCreatorGenerator.py]: Ok, fine for me.

**Comment by rhubert** [comment]
successful tested on several pc's. ready to merge from my side

**Pull request merged**